apps/chat/src/store/prompts/prompts.epics.ts

```diff
@@ -933,12 +933,32 @@ const uploadPromptsWithFoldersEpic: AppEpic = (action$) =>
         switchMap((foldersAndEntities) => {
           const folders = foldersAndEntities.flatMap((f) => f.folders);
           const prompts = foldersAndEntities.flatMap((f) => f.entities);
-          return of(
-            PromptsActions.uploadChildPromptsWithFoldersSuccess({
-              parentIds: payload.ids,
-              folders,
-              prompts,
-            }),
+          return concat(
+            of(
+              PromptsActions.uploadChildPromptsWithFoldersSuccess({
+                parentIds: payload.ids,
+                folders,
+                prompts,
+              }),
+            ),
+            iif(
+              () => !!payload.selectFirst,
+              concat(
+                of(
+                  PromptsActions.setSelectedPrompt({
+                    promptId: prompts[0].id,
+                  }),
+                ),
+                of(PromptsActions.uploadPrompt({ promptId: prompts[0].id })),
+                of(
+                  UIActions.openFolder({
+                    featureType: FeatureType.Prompt,
+                    id: prompts[0].folderId,
+                  }),
```

Comment:
this action will upload and open only one folder, but prompt can be several folders
e.g. if `folder1/folder2/folder3/prompt1` then you open only `folder3`